src/api/Intelligence.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @flow
+ * @file Helper for the Box AI API endpoint
+ * @author Box
+ */
+
+import Base from './Base';
+import { ERROR_CODE_INTELLIGENCE } from '../constants';
+import type { BoxItem } from '../common/types/core';
+import type { ElementsErrorCallback } from '../common/types/api';
+
+class Intelligence extends Base {
+    /**
+     * @property {Function}
+     */
+    successCallback: Function;
+
+    /**
+     * @property {Function}
+     */
+    errorCallback: ElementsErrorCallback;
+
+    /**
+     * API endpoint to ask ai a question
+     *
+     * @param {string} prompt - Question
+     * @param {Array<object>} items - Array of items to ask about
+     * @return {Promise}
+     */
+    async ask(prompt: string, items: Array<BoxItem>, successCallback: Function, errorCallback: ElementsErrorCallback) {
+        this.errorCode = ERROR_CODE_INTELLIGENCE;
+        if (!prompt) {
+            throw new Error('Missing prompt!');
+        }
+
+        if (!items || items.length === 0) {
+            throw new Error('Missing items!');
+        }
+
+        items.forEach(item => {
+            if (!item.id || !item.type) {
+                throw new Error('Invalid item!');
+            }
+        });
+
+        const url = `${this.getBaseApiUrl()}/ai/ask`;
+        const body = {
+            data: {
+                mode: 'single_item_qa',
+                prompt,
+                items,
+            },
+        };
+
+        this.successCallback = successCallback;
+        this.errorCallback = errorCallback;
+
+        await this.xhr
+            .post({
+                url,
+                id: items[0].id,
```

Comment:
Why is id = to the first items id?

Reply:
I honestly dont know why it even needs the id, its a requirement of the the xhr.post, but the items array will always have one item for now till this client is used for multi. items is required by the intelligence request it handles multi only

Reply:
looked it up, it seems to be using the id to get a writetoken to do whatever it needs to do

Reply:
yeah since the current mode we are using is 'single_item_qa', the array item will always has one item only.

Reply:
[nit] should we validate that items length is always 1?